### PR TITLE
Blog app - integration specs n+1 problems

### PR DESCRIPTION
**Pull request by mrkamin**
What does this PR do?

In this project, I will create integration tests for all of the views used in my project. 

Description of the task to be completed 👎

in this PR I will do bellow tasks 👎

- [x] Solving N+1 problem.

- Before:
N+1 quiry 1
N+1 quiry 2

- After:
N+1 quiry 3
N+1 quiry 4

- [x] Using Capybara to writing integration tests for each view in my project.

### CODE REVIEW

**Review by GzimAsani — CHANGES_REQUESTED** [review]
Hi @mrkamin , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that did not use [MIcroverse template](https://github.com/microverseinc/readme-template). Kindly add it.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

@GzimAsani,
thanks for you good and valuable feedback, 
your comments applied.

**Comment by addod19** [comment]
### Hi @mrkamin, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because your partner did not contribute to the project. Since this is a pair programming activity, we expect commits from both partners

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -29,4 +29,4 @@ def post_params
   end
 
   private :post_params
-end
```
Comment (Davidosky007) [inline app/controllers/posts_controller.rb:32]: * As mentioned in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=Make%20sure%20that%20the%20N%2B1%20problem%20is%20solved%20when%20fetching%20all%20posts%20and%20their%20comments%20for%20a%20user.) Kindly ensure that the N+1 problem is solved when fetching all posts and their comments for a user. You need to solve the n+1 issue in your code , HINT 💭 you can fetch all posts and comments belonging to a user by using this code `@posts = @user.posts.includes(:comments)`

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @mrkamin @AlejandraKeber 👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

* All tests passed 👍 
* Integration tests written properly 👍 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    @posts = Post.all.includes(:comments)
```
Comment (whiz25) [inline app/controllers/posts_controller.rb]: - Please use the User object to fetch all posts for that user as suggested earlier.
```ruby
@user = User.find(params[:user_id])
@posts = @user.posts.includes(:comments)
```

**Review by whiz25 — CHANGES_REQUESTED** [review]
Hi @mrkamin, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights


 ## Required Changes ♻️
_Check the comments under the review._
- Requested change made

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (mrkamin) [reply app/controllers/posts_controller.rb:32]: @Davidosky007 
thanks for your feedback,
your suggestion applied.

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (mrkamin) [reply app/controllers/posts_controller.rb]: @whiz25 
thanks for your feedback.
done

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @mrkamin and co,

✔️ Well-written and passing tests

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
@TedLivist 
thanks for your good and valuable feedback.

**Pull request merged**